### PR TITLE
Fix handling of libs of a different architecture

### PR DESCRIPTION
**Pull request by maxim-belkin**
@sjackman

### CODE REVIEW

File: Library/Homebrew/os/linux/elf.rb [hunk 1]
```diff
@@ -47,12 +47,14 @@ def initialize(path)
         @dylibs = []
         return
       end
-      command = ["ldd", path.expand_path.to_s]
-      libs = Utils.popen_read(*command).split("\n")
-      raise ErrorDuringExecution, command unless $?.success?
-      needed << "not found"
-      libs.select! { |lib| needed.any? { |soname| lib.include? soname } }
-      @dylibs = libs.map { |lib| lib[LDD_RX, 1] || lib[LDD_RX, 2] }.compact
+      if path.x86_64?
```
Comment (sjackman) [inline Library/Homebrew/os/linux/elf.rb]: I've created PR https://github.com/Linuxbrew/brew/pull/370 to add `Hardware::CPU.arch`.
I suggest
```diff
-if needed.empty?
+if needed.empty? || path.arch != Hardware::CPU::arch
   @dylibs = []
   return
 end
```

**Review by sjackman — CHANGES_REQUESTED** [review]

File: Library/Homebrew/os/linux/elf.rb (hunk 1 above)
Comment (maxim-belkin) [reply Library/Homebrew/os/linux/elf.rb]: agreed

File: Library/Homebrew/os/linux/elf.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@ def initialize(path)
       else
         odie "patchelf must be installed: brew install patchelf"
       end
-      if needed.empty?
+      if needed.empty? || path.arch != Hardware::CPU::arch
```
Comment (sjackman) [inline Library/Homebrew/os/linux/elf.rb]: I've merged PR https://github.com/Linuxbrew/brew/pull/370 for `Hardware::CPU.arch`

File: Library/Homebrew/extend/os/linux/keg_relocate.rb [hunk 3]
```diff
@@ -56,6 +56,7 @@ def detect_cxx_stdlibs(options = {})
     elf_files.each do |file|
       next if !file.dynamic? || file.mach_o_executable? && skip_executables
       dylibs = file.dynamically_linked_libraries
+      next if dylibs.nil?
```
Comment (sjackman) [inline Library/Homebrew/extend/os/linux/keg_relocate.rb]: Remove this line.

**Review by sjackman — CHANGES_REQUESTED** [review]

**Comment by sjackman** [comment]
Looks good to me. Does it fix the issue of pouring the `go` bottle?

File: Library/Homebrew/os/linux/elf.rb (hunk 2 above)
Comment (maxim-belkin) [reply Library/Homebrew/os/linux/elf.rb]: path.arch fails

**Comment by maxim-belkin** [comment]
> Looks good to me. Does it fix the issue of pouring the `go` bottle?

Yes, it does!

```
[linuxbrew@a2e874210168] [Tue May 02 19:27:47]
~/.linuxbrew/Library/Homebrew
(master) $ brew install go
==> Downloading https://linuxbrew.bintray.com/bottles/go-1.8.1.x86_64_linux.bottle.tar.gz
Already downloaded: /home/linuxbrew/.cache/Homebrew/go-1.8.1.x86_64_linux.bottle.tar.gz
==> Pouring go-1.8.1.x86_64_linux.bottle.tar.gz
Error: Failure while executing: ["ldd", "/home/linuxbrew/.linuxbrew/Cellar/go/1.8.1/libexec/src/debug/elf/testdata/gcc-386-freebsd-exec"] 

[linuxbrew@a2e874210168] [Tue May 02 19:28:07] [Last Exit Code: 1]
~/.linuxbrew/Library/Homebrew
(master) $ git checkout fix-32bit
Switched to branch 'fix-32bit'

[linuxbrew@a2e874210168] [Tue May 02 19:28:12]
~/.linuxbrew/Library/Homebrew
(fix-32bit) $ brew install go
==> Downloading https://linuxbrew.bintray.com/bottles/go-1.8.1.x86_64_linux.bottle.tar.gz
Already downloaded: /home/linuxbrew/.cache/Homebrew/go-1.8.1.x86_64_linux.bottle.tar.gz
==> Pouring go-1.8.1.x86_64_linux.bottle.tar.gz
==> Caveats
A valid GOPATH is required to use the `go get` command.
If $GOPATH is not specified, $HOME/go will be used by default:
  https://golang.org/doc/code.html#GOPATH

You may wish to add the GOROOT-based install location to your PATH:
  export PATH=$PATH:/home/linuxbrew/.linuxbrew/opt/go/libexec/bin

==> Summary
🍺  /home/linuxbrew/.linuxbrew/Cellar/go/1.8.1: 7,030 files, 287.0MB

[linuxbrew@a2e874210168] [Tue May 02 19:28:20]
~/.linuxbrew/Library/Homebrew
(fix-32bit) $
```

File: Library/Homebrew/os/linux/elf.rb (hunk 2 above)
Comment (maxim-belkin) [reply Library/Homebrew/os/linux/elf.rb]: disregard the above comment

**Review by sjackman — APPROVED** [review]

**Comment by sjackman** [comment]
Great! Once CI passes, go ahead and squash and merge.

**Comment by sjackman** [comment]
🚢

**Comment by sjackman** [comment]
Thanks, Maxim!

**Comment by maxim-belkin** [comment]
"Thanks" should go to you, Shaun :)

**Comment by sjackman** [comment]
Team effort!

**Pull request merged**